backend/de.metas.business/src/main/java/de/metas/project/budget/BudgetProjectRepository.java

```diff
@@ -23,28 +23,46 @@
 package de.metas.project.budget;
 
 import com.google.common.collect.ImmutableList;
+import de.metas.bpartner.BPartnerId;
+import de.metas.common.util.EmptyUtil;
 import de.metas.money.CurrencyId;
 import de.metas.organization.OrgId;
+import de.metas.pricing.PriceListVersionId;
 import de.metas.project.ProjectCategory;
 import de.metas.project.ProjectId;
+import de.metas.project.ProjectTypeId;
+import de.metas.project.workorder.data.ProjectQuery;
+import de.metas.user.UserId;
 import de.metas.util.InSetPredicate;
 import de.metas.util.Services;
+import de.metas.util.collections.CollectionUtils;
 import lombok.NonNull;
 import org.adempiere.ad.dao.IQueryBL;
+import org.adempiere.ad.dao.IQueryBuilder;
+import org.adempiere.exceptions.AdempiereException;
 import org.adempiere.model.InterfaceWrapperHelper;
 import org.compiere.model.I_C_Project;
+import org.compiere.util.TimeUtil;
 import org.springframework.stereotype.Repository;
 
 import java.util.List;
 import java.util.Objects;
 import java.util.Optional;
 
+import static org.adempiere.model.InterfaceWrapperHelper.saveRecord;
+
 @Repository
 public class BudgetProjectRepository
 {
-	final IQueryBL queryBL = Services.get(IQueryBL.class);
+	private final IQueryBL queryBL = Services.get(IQueryBL.class);
+	private final BudgetProjectResourceRepository budgetProjectResourceRepository;
```

Comment:
using a Repository from another Repository is usually a bad practice,
... and it's also a sign that:

Option 1. you might want to merge both repositories, BudgetProjectRepository and BudgetProjectResourceRepository, and consider BudgetProject as your aggregate root.
That means you are not allowed to retrieve `BudgetProjectResource` directly, but only by retrieving its aggregate root (BudgetProject).

Looking at BudgetProjectResourceRepository I think we don't want that because when we paint the visual calendar we have to fetch BudgetProjectResources from many projects.
So `BudgetProjectResource` it's an aggregate root on its own.

Option 2. Handle `BudgetProject` and `BudgetProjectResource` as two distinct aggregate roots (as we did until this change).
That means, 
* Each of BudgetProject and BudgetProjectResource shall have their own repository
* BudgetProject shall not contain BudgetProjectResource because from BudgetProject you are not able to load the BudgetProjectResource
* if you still need to have both of them somehow assembled (think about it, maybe you don't!), consider introducing BudgetProjectWithResources and have a service that queries both repositories and assembles your object.

wdyt?
cc @metas-ts

Reply:
i saw you marked this as resolved, but there is no change. the repo is still there.

Reply:
It was not used, I just forgot to remove it, removed now, thx